packages/calcite-components/src/utils/number.ts

```diff
@@ -217,3 +226,30 @@ export function expandExponentialNumberString(numberString: string): string {
 function stringContainsNumbers(string: string): boolean {
   return numberKeys.some((number) => string.includes(number));
 }
+
+/**
+ * Adds localized trailing decimals zero values to the number string.
+ * BigInt conversion to string removes the trailing decimal zero values (Ex: 1.000 is returned as 1). This method helps adding them back.
+ *
+ * @param {string} localizedValue - localized number string value
+ * @param {string} value - current value in the input field
+ * @param {NumberStringFormat} numberStringFormatter - numberStringFormatter instance to localize the number value
+ * @returns {string} localized number string value
+ */
+export function addLocalizedTrailingDecimalZeros(
+  localizedValue: string,
+  value: string,
+  formatter: NumberStringFormat
+): string {
+  const decimalSeparator = formatter.decimal;
+  const localizedIntegers = localizedValue.includes(decimalSeparator)
+    ? localizedValue.split(decimalSeparator)[0]
+    : localizedValue;
+  const decimals = value.split(".")[1];
+
+  return decimals
+    ? `${localizedIntegers}${decimalSeparator}${Array.from(decimals)
```

Comment:
It looks like the decimal separator is being added here as well as in input - 

```js
this.localizedValue =
        hasTrailingDecimalSeparator && isValueDeleted
          ? `${newLocalizedValue}${numberStringFormatter.decimal}`
```

Wouldn't that result in too many decimals?

Reply:
these are two diff scenarios, in the util method we add `decimalSeparator` only when we are appending decimal values. If there are decimal values then the condition `hasTrailingDecimalSeparator` return `false`.